src/common/BaseManager.ts

```diff
@@ -11,9 +12,59 @@ export type Props = {
     store: Store;
 };
 
-export default interface BaseManager {
+export interface Manager {
     destroy(): void;
     exists(parentEl: HTMLElement): boolean;
     render(props: Props): void;
     style(styles: Partial<CSSStyleDeclaration>): CSSStyleDeclaration;
 }
+
+export default class BaseManager implements Manager {
+    location: number;
+
+    reactEl: HTMLElement;
+
+    constructor({ location = 1, referenceEl }: Options) {
+        this.location = location;
+        this.reactEl = this.insert(referenceEl);
+
+        this.decorate();
+    }
+
+    decorate(): void {
```

Comment:
Introduced a new `decorate` method that subclasses need to implement to add their appropriate classname and testid